### PR TITLE
Allow for spec.Env to be set on the OTEL Colelctor Spec

### PR DESCRIPTION
**Pull request by ekarlso**
* Some of the extensions like Kubernetes Observer requires you to set some
  ENV vars
* Be useful for things like getting API Tokens via Vault etc.

### CODE REVIEW

File: pkg/collector/container.go [hunk 1]
```diff
@@ -63,5 +63,6 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 		Image:        image,
 		VolumeMounts: volumeMounts,
 		Args:         args,
+		Env:          otelcol.Spec.Env,
```
Comment (jpkrohling) [inline pkg/collector/container.go:66]: Could you add a test that exercises this?

**Review by jpkrohling — COMMENTED** [review]
Apart from the missing test, looks good to me.

File: pkg/collector/container.go (hunk 1 above)
Comment (ekarlso) [reply pkg/collector/container.go]: Yes I can look at it.

**Review by jpkrohling — APPROVED** [review]
LGTM

**Comment by jpkrohling** [comment]
Once you sort the EasyCLA check, it should be merged automatically by mergify. Thanks for the contribution!

**Pull request merged**